### DISCUSSION TITLE
👋 Welcoming ApplicationBuilder Feedback 👋

### DISCUSSION THREAD

**Discussion by lilyydu** [microsoft/teams-sdk#767] [Ideas / Feature requests]
Hello everyone! A few weeks ago, we introduced the `ApplicationBuilder` class as an alternative way to instantiate an `Application` instance. Now that some time has passed, we would love to hear any feedback and/or suggestions (especially if you've had the chance to use this in your development).

For those unfamiliar, the `ApplicationBuilder` provides greater readability and separates the management of configuration options (e.g., storage, turn state, AI module options, etc). This is available in both JS and C#, and examples of its use are shown in 2 samples (AI Message Extensions and Twenty Questions).

Right now, both approaches (`Application` and `ApplicationBuilder`) can be used, but we will eventually settle on one approach subject to feedback & developer experience. 

Whether you are migrating over, or new to building, we welcome all comments. Feel free to let us know if you have any questions/concerns.

Learn more in [JS](https://github.com/microsoft/teams-ai/blob/main/getting-started/js/00.MIGRATION.md#optional-applicationbuilder-class) and [C#](https://github.com/microsoft/teams-ai/blob/dotnetfluent/getting-started/dotnet/00.MIGRATION.md#optional-applicationbuilder-class) 
associated: #615 , #651 , #756

**Comment by Stevenic** [comment]
@lilyydu we should look into creating a similar builder for the various planners...
